model/project.go

```diff
@@ -866,12 +872,12 @@ func NewPatchTaskIdTable(proj *Project, v *Version, tasks TaskVariantPairs, proj
 	}
 	execTasksWithTaskGroupTasks := TVPairSet{}
 	for _, vt := range tasks.ExecTasks {
-		if _, ok := tgMap[vt.TaskName]; ok {
-			if tg := proj.FindTaskGroup(vt.TaskName); tg != nil {
-				for _, t := range tg.Tasks {
-					execTasksWithTaskGroupTasks = append(execTasksWithTaskGroupTasks, TVPair{vt.Variant, t})
-				}
+		bvt := proj.FindTaskForVariant(vt.TaskName, vt.Variant)
+		if tg := proj.FindTaskGroup(vt.TaskName); tg != nil {
```

Comment:
Sorry I'm noticing this so late. I'm slightly concerned about this; I think the purpose of having tgMap in the first place was to limit how often we need to loop through all of the task groups, and. this change would mean that for _every_ task we would loop through all task groups AND all tasks in all variants. Can we check tgMap and bvt.TaskGroup instead to avoid calling FindTaskGroup unless necessary?